R/as_mapper.R

```diff
@@ -62,35 +61,77 @@ as_function <- function(...) {
 
 #' Pluck out a single an element from a vector or environment
 #'
-#' This is a generalised form of `[[` which allows you to index by
-#' name, position, or attribute.
+#' @description
+#'
+#' This is a generalised form of `[[` which allows you to index deeply
+#' and flexibly into data structures. It supports R standard accessors
+#' like integer positions and string names and accepts any accessor
```

Comment:
.., and also accepts arbitrary accessor functions, functions that take an object and return some internal piece.